workflow-core/src/main/java/com/squareup/workflow1/RenderContext.kt

```diff
@@ -115,9 +100,159 @@ interface BaseRenderContext<out PropsT, StateT, in OutputT> {
     key: String,
     sideEffect: suspend () -> Unit
   )
+
+  /**
+   * Creates a function which builds a [WorkflowAction] from the
+   * given [update] function, and immediately passes it to [actionSink]. Handy for
+   * attaching event handlers to renderings.
+   *
+   * @param name A string describing the update, included in the action's [toString]
+   * as a debugging aid
+   * @param update Function that defines the workflow update.
+   */
+  fun eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>.Updater.() -> Unit
+  ): () -> Unit {
+    return {
+      actionSink.send(action(name, update))
```

Comment:
Why is `action` good enough here, but we need `NamedAction` for the others? Why do we need `NamedAction` at all, since `action{}` takes a name?

Reply:
I…don't remember. I thought I had a reason that I couldn't use action, but I'm not sure I actually believe that.